src/components/RouteChangeHandler/RouteChangeHandler.test.js

```diff
@@ -0,0 +1,154 @@
+  // a getBoundingClientRect method
+  document.querySelector = jest.fn(() => ({
+    getBoundingClientRect: () => ({
+      top: 137,
+    }),
+    style: {
+      outline: 5,
+    },
+    addEventListener: jest.fn(),
+    setAttribute: jest.fn(),
+    focus: jest.fn(),
+  }));
+
+  shallow(
+    <RouteChangeHandler
+      location={{ pathName: '/location-1' }}
+    >
+      The content
+    </RouteChangeHandler>
+  );
+
+  expect(document.querySelector).toHaveBeenCalledWith('#an-element-id');
+  expect(window.scrollTo).toHaveBeenCalledTimes(1);
+  expect(window.scrollTo).toHaveBeenCalledWith(0, 137 + 541);
+});
+
+it('should scroll on route change', () => {
+  const component = shallow(
+    <RouteChangeHandler
+      location={{ pathName: '/location-1' }}
+    >
+      The content
+    </RouteChangeHandler>
+  );
+
+  // it has scrolled once (on mount)
+  expect(window.scrollTo).toHaveBeenCalledTimes(1);
+
+  // change the props
+  component.setProps({
+    children: 'new children',
+  });
+
+  // but it doesn't try and scroll again since the location prop didn't change
+  expect(window.scrollTo).toHaveBeenCalledTimes(1);
+
+  // now change the location prop
+  component.setProps({
+    location: {
+      pathName: '/location-2',
+    }
+  });
+
+  // it should try and scroll again
+  expect(window.scrollTo).toHaveBeenCalledTimes(2);
+});
+
+it('should focus on an element when provided with a hash', () => {
+  document.location.hash = 'myhash';
+  const focusOnElement = jest.fn();
+
+  document.querySelector = jest.fn(() => ({
+    getBoundingClientRect: () => ({
+      top: 137,
+    }),
+    style: {
+      outline: 5,
+    },
+    addEventListener: jest.fn(),
+    focus: focusOnElement,
+    setAttribute: jest.fn(),
+  }));
+
+  const wrapper = mount(
+    <RouteChangeHandler location={{ pathname: '/1/2' }}>
+      <div id="myhash">The content</div>
+    </RouteChangeHandler>
+  );
+
+  wrapper.setProps({
+    location: {
+      pathname: '/2/1',
+    },
+  });
+
+  expect(document.querySelector).toHaveBeenCalledWith('#myhash');
+  expect(focusOnElement).toHaveBeenCalledTimes(1);
+});
+
+it('should focus on the specifed element when the path changes', () => {
+  document.location.hash = '';
+  const focusOnElement = jest.fn();
+  const focusElementId = ELEMENT_IDS.INITIAL_FOCUS_POINT_ON_PAGE;
+
+  document.getElementById = jest.fn(() => ({
+    getBoundingClientRect: () => ({
+      top: 137,
+    }),
+    style: {
+      outline: 5,
+    },
+    addEventListener: jest.fn(),
+    focus: focusOnElement,
+    setAttribute: jest.fn(),
+  }));
+
+  const wrapper = mount(
+    <RouteChangeHandler location={{ pathname: '/1/2' }}>
+      <div id={focusElementId}>The content</div>
+    </RouteChangeHandler>
+  );
+
+  wrapper.setProps({
+    location: {
+      pathname: '/2/1',
+    },
+  });
+
+  expect(document.getElementById).toHaveBeenCalledWith(focusElementId);
+  expect(focusOnElement).toHaveBeenCalledTimes(1);
+});
```

Comment:
Since every discovered bug should have a test, this should be testing that `tabIndex` and `outline` are reset on blur.

Reply:
Roger that